pkg/server/admin.go

```diff
@@ -3219,6 +3223,120 @@ func (s *systemAdminServer) SendKVBatch(
 	return br, nil
 }
 
+func (s *systemAdminServer) RecoveryCollectReplicaInfo(
+	request *serverpb.RecoveryCollectReplicaInfoRequest,
+	stream serverpb.Admin_RecoveryCollectReplicaInfoServer,
+) error {
+	ctx := stream.Context()
+	ctx = s.server.AnnotateCtx(ctx)
+	_, err := s.requireAdminUser(ctx)
+	if err != nil {
+		return err
+	}
+	log.Ops.Info(ctx, "streaming cluster replica recovery info")
+
+	return s.server.recoveryServer.ServeClusterReplicas(ctx, request, stream, s.server.db, s)
+}
+
+func (s *systemAdminServer) RecoveryCollectLocalReplicaInfo(
+	request *serverpb.RecoveryCollectLocalReplicaInfoRequest,
+	stream serverpb.Admin_RecoveryCollectLocalReplicaInfoServer,
+) error {
+	ctx := stream.Context()
+	ctx = s.server.AnnotateCtx(ctx)
+	_, err := s.requireAdminUser(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Ops.Info(ctx, "streaming local replica recovery info")
+	return s.server.recoveryServer.ServeLocalReplicas(ctx, request, stream)
+}
+
+func (s *systemAdminServer) RecoveryStagePlan(
+	ctx context.Context, request *serverpb.RecoveryStagePlanRequest,
+) (*serverpb.RecoveryStagePlanResponse, error) {
+	return nil, errors.AssertionFailedf("To be implemented by #93044")
+}
+
+func (s *systemAdminServer) RecoveryNodeStatus(
+	ctx context.Context, request *serverpb.RecoveryNodeStatusRequest,
+) (*serverpb.RecoveryNodeStatusResponse, error) {
+	return nil, errors.AssertionFailedf("To be implemented by #93043")
+}
+
+func (s *systemAdminServer) RecoveryVerify(
+	ctx context.Context, request *serverpb.RecoveryVerifyRequest,
+) (*serverpb.RecoveryVerifyResponse, error) {
+	return nil, errors.AssertionFailedf("To be implemented by #93043")
+}
+
+// VisitAvailableNodesWithRetry implements loqrecovery.ClusterAdminClient to
+// allow recovery module to perform operations on all cluster nodes.
+func (s *systemAdminServer) VisitAvailableNodesWithRetry(
+	ctx context.Context,
+	retryOpts retry.Options,
+	visitor func(nodeID roachpb.NodeID, client serverpb.AdminClient) error,
+) error {
+
+	collectNodeWithRetry := func(nodeID roachpb.NodeID) error {
+		var err error
+		for r := retry.StartWithCtx(ctx, retryOpts); r.Next(); {
+			log.Infof(ctx, "visiting node n%d, attempt %d", nodeID, r.CurrentAttempt())
+			var client serverpb.AdminClient
+			client, err = s.dialNode(ctx, nodeID)
```

Comment:
We can pass in the RPC context too, if we need it. Seems reasonable to use the same machinery that we normally use.

Reply:
Alternatively, make it a method on `loqrecovery.Server` and pass in gossip and the RPC context during construction.

Reply:
Yeah, I prefer the last idea of passing at construction as all the fields are static. But I think it makes more sense to make function a closure so that it could always be replaced for test. Otherwise it becomes more and more entangled with other components which is tough to break up for tests.